rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+                    if timestamp > to_timestamp:
+                        break
+                    operation = TroveOperation.deserialize(change['troveOperation'])
+                    collateral_change = deserialize_optional_fval(change['collateralChange'], 'collateralChange', 'liquity')  # noqa: E501
+                    debt_change = deserialize_optional_fval(change['debtChange'], 'debtChange', 'liquity')  # noqa: E501
+
+                    event = LiquityTroveEvent(
+                        kind='trove',
+                        tx=change['transaction']['id'],
+                        address=owner,
+                        timestamp=timestamp,
+                        debt_after=deserialize_optional_fval(change['debtAfter'], 'debtAfter', 'liquity'),  # noqa: E501
+                        collateral_after=deserialize_optional_fval(change['collateralAfter'], 'collateralAfter', 'liquity'),  # noqa: E501
+                        debt_delta=debt_change,
+                        collateral_delta=collateral_change,
+                        trove_operation=operation,
+                    )
+                    result[owner]['trove'].append(event)
+                except KeyError as e:
+                    self.msg_aggregator.add_warning(
+                        f'Ignoring Liquity Trove event in Liquity. '
+                        f'Failed to decode remote information. {str(e)}.',
+                    )
+                    continue
+
+        for stake in staked['lqtyStakes']:
+            owner = to_checksum_address(stake['id'])
+            for change in stake['changes']:
+                try:
+                    timestamp = change['transaction']['timestamp']
+                    if timestamp < from_timestamp:
+                        continue
+                    if timestamp > to_timestamp:
+                        break
+                    operation_stake = LiquityStakeEventType.deserialize(change['stakeOperation'])
+                    stake_event = LiquityStakeEvent(
+                        kind='stake',
+                        tx=change['transaction']['id'],
+                        address=owner,
+                        timestamp=timestamp,
+                        stake_after=deserialize_optional_fval(change['stakedAmountAfter'], 'stakedAmountAfter', 'liquity'),  # noqa: E501
+                        stake_change=deserialize_optional_fval(change['stakedAmountChange'], 'stakedAmountChange', 'liquity'),  # noqa: E501
+                        issuance_gain=deserialize_optional_fval(change['issuanceGain'], 'issuanceGain', 'liquity'),  # noqa: E501
+                        redemption_gain=deserialize_optional_fval(change['redemptionGain'], 'redemptionGain', 'liquity'),  # noqa: E501
+                        stake_operation=operation_stake,
+                    )
+                    result[owner]['stake'].append(stake_event)
+                except KeyError as e:
+                    self.msg_aggregator.add_warning(
```

Comment:
Add also a log warning with the actual entry so that we can inspect the logs to find the offending entry.